### PR TITLE
Fixes #29558 - use Rails 6 on Ruby >= 2.6

### PR DESCRIPTION
**Pull request by mmoll**

### CODE REVIEW

**Comment by mmoll** [comment]
Needs to wait for a newer PostgreSQL on the Jenkins slaves.

File: test/controllers/api/v2/organizations_controller_test.rb [hunk 1]
```diff
@@ -113,7 +113,12 @@ class Api::V2::OrganizationsControllerTest < ActionController::TestCase
   end
 
   test "should not create with content type text plain" do
-    assert_raises_with_message(RuntimeError, 'Unknown Content-Type') do
+    if Rails.version.start_with?('5.2')
+      error_message = 'Unknown Content-Type'
+    else
+      error_message = '"" is not a valid MIME type'
```
Comment (tbrisker) [inline test/controllers/api/v2/organizations_controller_test.rb]: Rails 6 raises `InvalidMimeType`:
https://github.com/rails/rails/blob/6aad3b01506677d134fb4512b31177c44aa9ee77/actionpack/lib/action_dispatch/http/mime_type.rb#L238
Should we just drop the message and check for the correct exception type?

Comment (mmoll) [reply test/controllers/api/v2/organizations_controller_test.rb]: I'd expect that message to state the wrong mimetype, so this need still some digging.

But this commit is only a hack to get things green anyway.

**Comment by mmoll** [comment]
Indeed green with PG 12. :green_heart:

**Comment by amirfefer** [comment]
After a deep investigation about graphql test failures, I assume the reason is `ActiveRecord::Associations::Preloader` which we use [here](https://github.com/theforeman/foreman/blob/8033f112afa5aaf45ce8d970b6f491bf7052feee/app/graphql/collection_loader.rb#L41)
The given `base_scope` is similar, but the results after the preloading are not as expected with rails 6. 

I searched for an issue in rails repo, and I found [this one](https://github.com/rails/rails/issues/38419) and some more.   
 
Also as far as I understand, `ActiveRecord::Associations::Preloader` is for internal use and isn't exposed as a public API.

**Comment by tbrisker** [comment]
[test foreman]

**Comment by ezr-ondrej** [comment]
@mmoll could you rebase and reenable the graphql tests?

File: test/controllers/api/v2/provisioning_templates_controller_test.rb [hunk 2]
```diff
@@ -126,7 +126,11 @@ class Api::V2::ProvisioningTemplatesControllerTest < ActionController::TestCase
     assert_response :success
     assert_equal 'text/plain', response.content_type
     assert_equal templates(:pxekickstart).to_erb, response.body
-    assert_equal 'attachment; filename="centos5_3_pxelinux.erb"', response.headers['Content-Disposition']
+    if Rails.version.start_with?('5.2')
+      assert_equal 'attachment; filename="centos5_3_pxelinux.erb"', response.headers['Content-Disposition']
+    else
+      assert_equal 'attachment; filename="centos5_3_pxelinux.erb"; filename*=UTF-8\'\'centos5_3_pxelinux.erb', response.headers['Content-Disposition']
```
Comment (ezr-ondrej) [inline test/controllers/api/v2/provisioning_templates_controller_test.rb]: is there a point to assert exact equality? what if we 
```suggestion
      assert_match /attachment; filename="centos5_3_pxelinux.erb"/
```
we don't need the if then :thinking:

**Comment by mmoll** [comment]
Looks all green and ready to me :smiley:

**Review by ezr-ondrej — APPROVED** [review]
Thanks @mmoll :green_heart: !
Let's do this!

**Pull request merged**